src/javascripts/ng-admin/es6/lib/View/View.js

```diff
@@ -0,0 +1,289 @@
+        return this;
+    }
+
+    getReferences() {
+        var references = {};
+        var referenceFields = this._fields.filter(field => field.type() === 'reference' || field.type() === 'reference_many');
+        for (var key in referenceFields) {
+            let referencedField = referenceFields[key];
+            references[referencedField.name()] = referencedField;
+        }
+
+        return references;
+    }
+
+    getReferencedLists() {
+        var result = {};
+        var lists = this._fields.filter(f => f.type() === 'referenced_list');
+        for (let i = 0, c = lists.length ; i < c ; i++) {
+            let list = lists[i];
+            result[list.name()] = list;
+        }
+
+        return result;
+    };
+
+    mapEntry(restEntry) {
+        return new Entry.mapFromRest(this, restEntry);
+    }
+
+    mapEntries(restEntries) {
+        return restEntries.map(e => this.mapEntry(e));
+    }
+
+    template(template) {
+        if (!arguments.length) {
+            return this._template;
+        }
+
+        this._template = template;
+
+        return this;
+    }
+
+    identifier() {
+        var identifier;
+
+        var fields = this._fields;
+        for (var i in fields) {
+            if (fields[i].identifier()) {
+                identifier = fields[i];
+                break;
+            }
+        }
+
+        // No identifier fields on this view, try to find it on other view
+        if (!identifier) {
+            identifier = this.entity.identifier();
+        }
+
+        if (!arguments.length) {
+            return identifier;
+        }
+
+        return this;
+    }
+
+    actions(actions) {
+        if (!arguments.length) return this._actions;
+        this._actions = actions;
+        return this;
+    }
+
+    processFieldsDefaultValue(entry) {
+        for (var i in this._fields) {
+            var field = this._fields[i];
+            entry.values[field.name()] = field.defaultValue();
+        }
+
+        return this;
+    }
+
+    removeFields() {
+        this._fields = [];
+        return this;
+    }
+
+    getFields(asArray) {
+        if (asArray) {
+            return this._fields;
+        }
+
+        return View._indexFieldsByName(this._fields);
+    }
+
+    getField(fieldName) {
+        return this._fields.filter(f => f.name() === fieldName)[0];
+    }
+
+    getFieldsOfType(type) {
+        var fields = this._fields.filter(f => f.type() === type);
+        return View._indexFieldsByName(fields);
+    }
+
+    static _indexFieldsByName(fields) {
+        var result = {};
+        for(let i = 0, c = fields.length ; i < c ; i++) {
+            var field = fields[i];
+            result[field.name()] = field;
+        }
+
+        return result;
+    }
+
+    addField(field) {
+        if (field.order() === null) {
+            field.order(this._fields.length);
+        }
+
+        this._fields.push(field);
+        return this;
+    }
+
+    getErrorMessage(response) {
```

Comment:
Can't find any simple solution at the moment. I'll perhaps do it in another PR, but not in this huge one.